src/Sulu/Bundle/AdminBundle/Resources/js/components/TextArea/tests/TextArea.test.js

```diff
@@ -42,32 +43,35 @@ test('TextArea should render with value and character counter', () => {
     expect(container).toMatchSnapshot();
 });
 
-test('TextArea should call onBlur when it loses focus', () => {
+test('TextArea should call onBlur when it loses focus', async() => {
     const blurSpy = jest.fn();
     render(<TextArea onBlur={blurSpy} onChange={jest.fn()} value="" />);
 
     const textarea = screen.queryByRole('textbox');
-    fireEvent.blur(textarea);
 
+    await userEvent.click(textarea);
+    expect(blurSpy).not.toBeCalledWith();
+
+    await userEvent.tab();
     expect(blurSpy).toBeCalledWith();
 });
 
-test('TextArea should call onChange when the TextArea changes', () => {
+test('TextArea should call onChange when the TextArea changes', async() => {
     const changeSpy = jest.fn();
     render(<TextArea onChange={changeSpy} value="My value" />);
 
     const textarea = screen.queryByDisplayValue('My value');
-    fireEvent.change(textarea, {target: {value: 'my-value'}});
+    await userEvent.type(textarea, '!');
```

Comment:
when typing something to the component, the `changeSpy` is called after every character. we do not update the `value` that is passed to the component (`value="My value"`) when our `changeSpy` is called.
unfortunately, this means we cannot type more than one character without rewriting the tests 😕

Reply:
okay lets see how this works. Else there are 2 solutions our Input work out of the box via internal state which would I think break our "source of truth" or our tests us a "test component".

Reply:
The fact that only one character can be typed leads sometimes to problems or unexpected behavior (at least for me). E. g. when working with floating number inputs one necessarily needs to type 3 characters to test it. `1`->`1.`->`1.5` otherwise

Reply:
i have implemented a helper function for wrapping controlled components in https://github.com/sulu/sulu/pull/6795. this should hopefully solve such problems 🙂